lib/p2p/Pool.ts

```diff
@@ -411,18 +413,46 @@ class Pool extends EventEmitter {
     if (!deal) {
       return;
     }
-    if (!deal.r_preimage) {
-      // TODO: Handle this unexpected case, if we are the maker we should have the preimage.
-      return;
-    }
-    const body: packets.SwapResponsePacketBody = {
-      r_preimage: deal.r_preimage,
-    };
-    const swapResponsePacket = new packets.SwapResponse(body, requestPacket.header.id);
 
-    this.logger.debug('sending back to peer: '  + JSON.stringify(body));
+    let cmdLnd = this.lndLtcClient;
+
+  	switch (deal.makerCurrency) {
+    case 'BTC':
+      break;
+    case 'LTC':
+      cmdLnd = this.lndBtcClient;
+      break;
+  	}
+  	if (!cmdLnd) {
+  	  return;
+  	}
+
+    const request = new lndrpc.SendRequest();
+  	request.setAmt(deal.takerAmount);
+  	request.setDestString(deal.takerPubKey);
+  	request.setPaymentHashString(String(deal.r_hash));
+
+    // TODO: use timeout on call
+
+    try {
+      const response = await cmdLnd.sendPaymentSync(request);
+      if (response.getPaymentError()) {
+        this.logger.error('Got error from sendPaymentSync: ' + response.getPaymentError() + ' ' + JSON.stringify(request.toObject()));
+        return;
+      }
+
+      const body: packets.SwapResponsePacketBody = {
+        r_preimage: Buffer.from(response.getPaymentPreimage_asB64(), 'base64').toString('hex'),
+      };
+      const swapResponsePacket = new packets.SwapResponse(body, requestPacket.header.id);
+
+      this.logger.debug('sending back to peer: '  + JSON.stringify(body));
+
+      peer.sendPacket(swapResponsePacket);
 
-    peer.sendPacket(swapResponsePacket);
+    } catch (err) {
+      this.logger.error('Got exception from sendPaymentSync: ' + ' ' + JSON.stringify(request.toObject()));
```

Comment:
This should log the `err` object, there should be an `err.message` and `err.code`.

Reply:
How can you be sure? This is why I don't like this mechanism.

Change as part of refactor

Reply:
You can actually just append the `err` as a second parameter to the `logger.error` function, that should work too, but I'll handle it. Just need to have some sort of indicator of what the error was.